app/code/Magento/CatalogInventory/Api/StockStateInterface.php

```diff
@@ -27,13 +27,14 @@ public function verifyNotification($productId, $scopeId = null);
 
     /**
      * @param int $productId
+     * @parem $product
      * @param float $itemQty
      * @param float $qtyToCheck
      * @param float $origQty
      * @param int $scopeId
      * @return int
      */
-    public function checkQuoteItemQty($productId, $itemQty, $qtyToCheck, $origQty, $scopeId = null);
+    public function checkQuoteItemQty($productId, $product, $itemQty, $qtyToCheck, $origQty, $scopeId = null);
```

Comment:
Fix is not correct - you cannot change Api methods like this.

Since you already have a `productId` within method, for a proper fix you need to find out why `name` is not loaded correctly in these circumstances.